### PR TITLE
Following moving elements across the document

### LINKED ISSUE #49: Sometimes misses element
I can't reproduce it, but I have seen this library missing an element. The click apparently hit another element next to it. 99% of the time it seems to work though.

### PR DESCRIPTION
**Pull request by Xetera**
On sites with high CLS scores, the element that's being clicked on will sometimes miss due to page reflow changing the position of the target between the time its coordinates are read, and the mouse moving to its location.

This change makes the `move` function check whether it's overlapping with the coordinates of the element being clicked on before moving on to perform the click. It will retry to move to its new location up to a configurable amount of times before throwing an error.

https://user-images.githubusercontent.com/24978328/170891010-f711a5d0-3723-4f25-af21-24634e458326.mov

It also includes:
* Groundwork for unit tests with jest-puppeteer + swc
* Some formatting fixes with `npx ts-standard --fix`
* Edge case fix with `mouse-helper` where it wouldn't attach to a page that already loaded (specifically when using `page.setContent`)

Closes #49

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
```
Comment (Niek) [inline jest.config.js:1]: What about moving the config to `package.json`? That saves another file (2?) in the repo: https://jestjs.io/docs/configuration

**Review by Niek — COMMENTED** [review]
Works great :) I do get an error with linting though:
```console
$ yarn lint
yarn run v1.22.19
$ yarn ts-standard --fix
ts-standard: Typescript Standard Style! (https://github.com/standard/ts-standard)
  src/__test__/spoof.spec.ts:0:0: Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: src/__test__/spoof.spec.ts.
The file must be included in at least one of the projects provided. ()
```

**Comment by Xetera** [comment]
Yeah I'm really not sure what that's about... I linted the entire thing with `yarn ts-standard --fix` locally and it doesn't throw any errors for me. Do I need to specify the test file somewhere for ts-standard?

File: jest.config.js (hunk 1 above)
Comment (Xetera) [reply jest.config.js:1]: Oh that's a good idea, I'll do that

**Comment by Niek** [comment]
> Yeah I'm really not sure what that's about... I linted the entire thing with `yarn ts-standard --fix` locally and it doesn't throw any errors for me. Do I need to specify the test file somewhere for ts-standard?

I fixed it now in https://github.com/Xetera/ghost-cursor/pull/54/commits/d75ca3e7f747ae0b720b46b506c05ce6f05d483b

**Review by Niek — APPROVED** [review]

**Comment by Xetera** [comment]
Ah great, thanks a lot
